File: Data Structures & Algorithms/anagram-groups/submission-7.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hsh;
        for (string word : strs) {
            string wrd = word;
            sort(wrd.begin(), wrd.end());
            hsh[wrd].push_back(word);
        }
        vector<vector<string>> res;
        for (const auto& [key, arr] : hsh) {
            res.push_back(arr);
        }
        return res;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-7.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        return "".join([f"{len(word)}#{word}" for word in strs])

    def decode(self, s: str) -> List[str]:
        res = []
        i = 0
        while i < len(s):
            # find num
            j = i + 1
            while s[j] != "#":
                j += 1
            num = int(s[i:j])
            # find word
            start = j + 1
            end = start + num
            res.append(s[start:end])
            i = end
        return res
